internal/csi-addons/networkfence/fencing.go

```diff
@@ -132,6 +145,149 @@ func (nf *NetworkFence) AddNetworkFence(ctx context.Context) error {
+	_, stdErr, err := util.ExecCommandWithTimeout(ctx, 5*time.Minute, "ceph", cmd...)
+	if err != nil {
+		return fmt.Errorf("failed to evict client %d: %w, stderr: %q", clientID, err, stdErr)
+	}
+	log.DebugLog(ctx, "client %s has been evicted from CephFS\n", clientID)
+
+	return nil
+}
+
+func isIPInCIDR(ctx context.Context, ip, cidr string) bool {
+	// Parse the CIDR block
+	_, ipCidr, err := net.ParseCIDR(cidr)
+	if err != nil {
+		log.ErrorLog(ctx, "error parsing CIDR block %s: %w\n", cidr, err)
+
+		return false
+	}
+
+	// Parse the IP address
+	ipAddress := net.ParseIP(ip)
+	if ipAddress == nil {
+		log.ErrorLog(ctx, "error parsing IP address %s\n", ip)
+
+		return false
+	}
+
+	// Check if the IP address is within the CIDR block
+	return ipCidr.Contains(ipAddress)
+}
+
+func (ac *activeClient) fetchIP() (string, error) {
+	// example: "inst": "client.4305 172.21.9.34:0/422650892",
+	// then returning value will be 172.21.9.34
+	clientInfo := ac.Inst
+	parts := strings.Fields(clientInfo)
+	if len(parts) >= 2 {
+		ip := strings.Split(parts[1], ":")[0]
+
+		return ip, nil
+	}
+
+	return "", errors.New("failed to extract IP address")
+}
+
+func (ac *activeClient) fetchID() (int, error) {
+	// example: "inst": "client.4305 172.21.9.34:0/422650892",
+	// then returning value will be 4305
+	clientInfo := ac.Inst
+	parts := strings.Fields(clientInfo)
+	if len(parts) >= 1 {
+		clientIDStr := strings.TrimPrefix(parts[0], "client.")
+		clientID, err := strconv.Atoi(clientIDStr)
+		if err != nil {
+			return 0, fmt.Errorf("failed to convert client ID to int: %w", err)
+		}
+
+		return clientID, nil
+	}
+
+	return 0, fmt.Errorf("failed to extract client ID, incorrect format: %s", clientInfo)
+
+}
+
+// AddClientEviction blocks access for all the IPs in the CIDR block
+// using client eviction.
+// blocks the active clients listed in cidr, and the IPs
+// for whom there is no active client present too.
+func (nf *NetworkFence) AddClientEviction(ctx context.Context) error {
+	evictedIPs := make(map[string]bool)
+	// fetch active clients
+	activeClients, err := listActiveClients(ctx)
+	if err != nil {
+		return err
+	}
+	// iterate through CIDR blocks and check if any active client matches
+	for _, cidr := range nf.Cidr {
+		for _, client := range activeClients {
+			clientIP, err := client.fetchIP()
+			if err != nil {
+				return fmt.Errorf("error fetching client IP: %w", err)
+			}
+			// check if the clientIP is in the CIDR block
+			if isIPInCIDR(ctx, clientIP, cidr) {
+				clientID, err := client.fetchID()
+				if err != nil {
+					return fmt.Errorf("error fetching client ID: %w", err)
+				}
+				// evict the client
+				err = evictCephFSClient(ctx, clientID)
+				if err != nil {
+					return fmt.Errorf("error evicting client %d: %w", clientID, err)
+				}
+				log.DebugLog(ctx, "client %d has been evicted\n", clientID)
+				// add the CIDR to the list of blocklisted IPs
+				evictedIPs[cidr] = true
```

Comment:
You need to track the singe client IP-address that was evicted, not the whole CIDR block. There might be other IP-addresses in the CIDR block that are not evicted, and those will need to be blocklisted.

Instead of `cidr` as key in the map, use `client.fetchIP()`.